engine.py

```diff
@@ -489,6 +489,18 @@ def _run_app_instance_commands(self):
         # to houdini as an event loop callback. This will run when houdini is
         # idle which should be after the UI loads up.
         def run_when_idle():
+            # We have a problem in H16 where the event loop callback is
+            # being called too early. We want to wait, but we don't want
+            # to block. There are a few options on how to make that happen,
+            # but rather than do something time based, it's acting more reliable
+            # when we wait some number of event loop iterations before completing
+            # our job. Going with 10 seems to get us past the threshold reliably
+            # without it being noticeable to the user.
+            if run_when_idle._call_count < 10:
```

Comment:
Seems odd to rely on an arbitrary value like this. Is there a way to test that the UI is available here? Something like checking `hou.qt.mainWindow()`? Not sure if that would work. If it did, I guess we'd want a timeout too.